lightning/src/ln/channel.rs

```diff
@@ -609,11 +609,11 @@ impl<Signer: Sign> Channel<Signer> {
 			counterparty_dust_limit_satoshis: 0,
 			holder_dust_limit_satoshis: MIN_DUST_LIMIT_SATOSHIS,
 			counterparty_max_htlc_value_in_flight_msat: 0,
-			counterparty_selected_channel_reserve_satoshis: 0,
+			counterparty_selected_channel_reserve_satoshis: None,
 			counterparty_htlc_minimum_msat: 0,
 			holder_htlc_minimum_msat: if config.own_channel_config.our_htlc_minimum_msat == 0 { 1 } else { config.own_channel_config.our_htlc_minimum_msat },
 			counterparty_max_accepted_htlcs: 0,
-			minimum_depth: 0, // Filled in in accept_channel
+			minimum_depth: None, // Filled in in accept_channel
```

Comment:
Would be nice if `Channel` had a state that held this data rather than knowing to call `unwrap` when we know it is safe. But I guess there's a lot more work to get there. 😕

Reply:
Yea, I've thought a bit before about inlining a ton of state variables inside `ChannelState` or something similar, but as you note its a lot of change. At least these unwraps are pretty trivial - we don't have to be very far along in the state machine for them to be known-good.